spec/domain/people/duplicate_locator_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2012-2020, Jungwacht Blauring Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+require 'spec_helper'
+
+describe People::DuplicateLocator do
+
+  let(:checker) { described_class.new }
+
+  context 'check' do
+    it 'creates one duplicate when all attributes match once' do
+      create_duplicate(:top_leader)
+      expect { checker.run }.to change { PersonDuplicate.count }.by(1)
+    end
+
+    it 'creates two duplicate when all attributes match twice' do
+      create_duplicate(:top_leader)
+      create_duplicate(:top_leader)
+      expect { checker.run }.to change { PersonDuplicate.count }.by(2)
+    end
+
+    it 'creates one duplicate for matching name, company name, birthday but blank zip code' do
+      create_duplicate(:top_leader).update!(zip_code: nil)
+      expect { checker.run }.to change { PersonDuplicate.count }.by(1)
+    end
+
+    it 'creates one duplicate for matching name, company name, zip code but blank birthday' do
+      create_duplicate(:top_leader).update!(birthday: nil)
+      expect { checker.run }.to change { PersonDuplicate.count }.by(1)
+    end
+
+    it 'creates one duplicate for matching name, company name but blank birthday, zip code' do
+      create_duplicate(:top_leader).update!(birthday: nil, zip_code: nil)
+      expect { checker.run }.to change { PersonDuplicate.count }.by(1)
+    end
+
+    it 'creates no duplicate when duplicate already exists' do
+      duplicate = create_duplicate(:top_leader)
+      PersonDuplicate.create(person_1: duplicate, person_2: people(:top_leader))
```

Comment:
create!

if for any reason it can't be created ... we get noticed that way. 

in this case the test would probably fail here ... but it's good practice to always use the bang operator unless you do error handling or don't care about when creation or update fails